File: codeforces/virtual-contest-div-4/insane.py
def find_output(k, l1, r1, l2, r2):
    count = 0
    for i in range(l1, r1 + 1):
        for j in range(i, r2 + 1):
            val = j / i 
            if k % val == 0:
                count += 1
    return count

num_tests = int(input())
output = []
for _ in range(num_tests):
    k, l1, r1, l2, r2 = map(int, input().split())
    output.append(find_output(k, l1, r1, l2, r2))

for val in output:
    print(val)

File: codeforces/contests/993/insane.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int findY(int k, int l1, int r1, int l2, int r2) {
    int count = 0;
    long long kn=1;
    for (int n = 0; r2/kn>=l1; n++) {
        int min_x = max(l1, (int)((l2 + kn - 1) / kn));
        int max_x = min(r1, (int)(r2 / kn));
        if (max_x >= min_x) count += max_x - min_x + 1;
        kn *= k;
    }
    return count;
}

int main() {
    int t;
    cin >> t;
    vector<int> res;
    for(int i = 0; i < t; i++) {
        vector<int> input(5);
        for (int &x: input) cin>>x;
        int k = input[0];
        int l1 = input[1];
        int r1 = input[2];
        int l2 = input[3];
        int r2 = input[4];
        int output = findY(k, l1, r1, l2, r2);
        res.push_back(output);
    }
    for (int val: res) {
        cout << val << endl;
    }
}